CMakeLists.txt

```diff
@@ -44,22 +44,38 @@ if(APPLE)
   set(CMAKE_CXX_FLAGS "-stdlib=libc++")
 endif()
 
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-m64 -pipe -O2 -Wall -W -fPIE")
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-m64 -pipe -fno-strict-aliasing -frounding-math -std=c++0x -O2 -std=c++0x -Wall -W -Wno-unused-local-typedefs -fPIE")
```

Comment:
These look like gcc flags and should be wrapped in an appropriate check.
Also, m64 should probably be enabled only for 64-bit builds

Reply:
I fixed this as you suggested (and also all the typos you mentioned above)